notebook/notebookapp.py

```diff
@@ -1156,7 +1157,7 @@ def init_server_extensions(self):
             if not modulename in self.nbserver_extensions:
                 self.nbserver_extensions[modulename] = True
         
-        for modulename in self.nbserver_extensions:
+        for modulename in sorted(self.nbserver_extensions.items(), key=lambda k,v:k):
```

Comment:
Why do you need the full `.items()` call? Since you're only iterating on keys, as far as I can tell, a simple `sorted(self.nbserver_extensions)` should suffice...